sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -403,7 +403,11 @@ public Expression field(BlockBuilder list, int index, Type storageType) {
       } else if (fromType.getTypeName().isCompositeType()
           || (fromType.getTypeName().isCollectionType()
               && fromType.getCollectionElementType().getTypeName().isCompositeType())) {
-        field = Expressions.call(WrappedList.class, "of", field);
+        field =
```

Comment:
Wouldn't this fix be applicable to other types (logical and datetime) as well? I think they can be nullable

Reply:
Yes. It is applicable. 
I will try to do the changes and upload. Need to add UT for the same.